code/modules/shuttle/shuttle.dm

```diff
@@ -324,54 +360,49 @@
 			Ts1.copy_air_with_tile(T0)
 
 		//move mobile to new location
-		loc = S1.loc
-		dir = S1.dir
 
-		//move all objects
-		for(var/obj/O in T0)
-			if(O.invisibility >= 101)
-				continue
-			if(O == T0.lighting_object)
-				continue
-			O.loc = T1
 
-			//close open doors
-			if(istype(O, /obj/machinery/door))
-				var/obj/machinery/door/Door = O
-				spawn(-1)
-					if(Door)
-						Door.close()
 
-		for(var/mob/M in T0)
-			if(!M.move_on_shuttle)
-				continue
-			M.loc = T1
-
-			//docking turbulence
-			if(M.client)
-				spawn(0)
-					if(M.buckled)
-						shake_camera(M, 2, 1) // turn it down a bit come on
-					else
-						shake_camera(M, 7, 1)
-			if(istype(M, /mob/living/carbon))
-				if(!M.buckled)
-					M.Weaken(3)
-
-		T0.ChangeTurf(turf_type)
+		for(var/atom/movable/AM in T0)
+			if (rotation)
+				AM.shuttleRotate(rotation)
 
-	//air system updates
-	for(var/turf/T1 in L1)
-		T1.redraw_lighting()
-		SSair.remove_from_active(T1)
-		T1.CalculateAdjacentTurfs()
-		SSair.add_to_active(T1,1)
+			if (istype(AM,/obj))
+				var/obj/O = AM
+				if(O.invisibility >= 101)
+					continue
+				if(O == T0.lighting_object)
+					continue
+				O.loc = T1
+
+				//close open doors
+				if(istype(O, /obj/machinery/door))
+					var/obj/machinery/door/Door = O
+					spawn(-1)
+						if(Door)
+							Door.close()
+			else if (istype(AM,/mob))
+				var/mob/M = AM
+				if(!M.move_on_shuttle)
+					continue
+				M.loc = T1
+
+				//docking turbulence
+				if(M.client)
+					spawn(0)
+						if(M.buckled)
+							shake_camera(M, 2, 1) // turn it down a bit come on
+						else
+							shake_camera(M, 7, 1)
+				if(istype(M, /mob/living/carbon))
+					if(!M.buckled)
+						M.Weaken(3)
 

```

Comment:
Somewhere around here add this loop:

```
for(var/turf/T in L1)
		for(var/atom/A in T)
			if(A.smooth)
				smooth_icon(A)
```

This will resmooth all smooth atoms *after* they're moved. Smoothing onboard the shuttle was working before because of the spawn in the smooth code, which was removed. This is needed now.